sw/host/tests/chip/spi_passthru/src/main.rs

```diff
@@ -442,17 +486,89 @@ fn test_read_flash(opts: &Opts, transport: &TransportWrapper, mode: ReadMode) ->
 fn main() -> Result<()> {
     let opts = Opts::parse();
     opts.init.init_logging();
+    /* Load the ELF binary and get the address of the `kReadPipelineMode` variable
+     * in example_sival.c */
+    let elf_binary = fs::read(&opts.firmware_elf)?;
+    let elf_file = object::File::parse(&*elf_binary)?;
+    let mut symbols = HashMap::<String, u32>::new();
+    for sym in elf_file.symbols() {
+        symbols.insert(sym.name()?.to_owned(), sym.address() as u32);
+    }
+    let pipeline_mode_address = symbols
+        .get("kReadPipelineMode")
+        .expect("Provided ELF missing 'kReadPipelineMode' symbol");
+
     let transport = opts.init.init_target()?;
 
     let uart = transport.uart("console")?;
     uart.set_flow_control(true)?;
-    let _ = UartConsole::wait_for(&*uart, r"Running [^\r\n]*", opts.timeout)?;
+    let _ = UartConsole::wait_for(&*uart, r"Status1 is [^\r\n]*", opts.timeout)?;
```

Comment:
Oops, didn't mean to keep this change. We're not handling the QE bit in this test yet.